src/QueueFactory.php

```diff
@@ -74,17 +72,16 @@ public function get(string $channel = self::DEFAULT_CHANNEL_NAME): QueueInterfac
     /**
      * @param string $channel
      *
-     * @throws InvalidConfigException
+     * @throws ChannelIncorrectlyConfigured
      *
      * @return QueueInterface
      */
     private function create(string $channel): QueueInterface
     {
         if (isset($this->channelConfiguration[$channel])) {
-            $adapter = $this->yiiFactory->create($this->channelConfiguration[$channel]);
-            if (!$adapter instanceof AdapterInterface) {
-                throw new ChannelIncorrectlyConfigured($channel, $adapter);
-            }
+            $definition = $this->channelConfiguration[$channel];
+            $this->checkDefinitionType($channel, $definition);
+            $adapter = $this->createFromDefinition($channel, $definition);
```

Comment:
```suggestion
            $definition = Normalizer::normalize($this->channelConfiguration[$channel]);
            $adapter = $definition->resolve($this->container);
            if (!$adapter instanceof AdapterInterface) {
                throw new ChannelIncorrectlyConfigured($channel, $adapter);
            }
```

Seems, it's code simplify and work good also.

Reply:
This doesn't work for `[ClassName::class, 'methodName']` definitions. It's Yii super-power, I'd like to have it everywhere in callable definitions.